geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/LoadBalanceGatewaySenderCommand.java

```diff
@@ -25,70 +25,74 @@
       help = CliStrings.LOAD_BALANCE_GATEWAYSENDER__HELP)
   @CliMetaData(relatedTopic = CliStrings.TOPIC_GEODE_WAN)
   @ResourceOperation(resource = ResourcePermission.Resource.CLUSTER,
       operation = ResourcePermission.Operation.MANAGE, target = ResourcePermission.Target.GATEWAY)
-  public Result loadBalanceGatewaySender(@CliOption(key = CliStrings.LOAD_BALANCE_GATEWAYSENDER__ID,
-      mandatory = true, optionContext = ConverterHint.GATEWAY_SENDER_ID,
-      help = CliStrings.LOAD_BALANCE_GATEWAYSENDER__ID__HELP) String senderId) {
+  public ResultModel loadBalanceGatewaySender(
+      @CliOption(key = CliStrings.LOAD_BALANCE_GATEWAYSENDER__ID,
+          mandatory = true, optionContext = ConverterHint.GATEWAY_SENDER_ID,
+          help = CliStrings.LOAD_BALANCE_GATEWAYSENDER__ID__HELP) String senderId) {
 
-    Result result;
     if (senderId != null) {
       senderId = senderId.trim();
     }
 
     Cache cache = getCache();
-    SystemManagementService service = (SystemManagementService) getManagementService();
-    TabularResultData resultData = ResultBuilder.createTabularResultData();
-    Set<DistributedMember> dsMembers = getAllNormalMembers();
+    SystemManagementService service =
+        (SystemManagementService) ManagementService.getExistingManagementService(getCache());
 
+    Set<DistributedMember> dsMembers = getAllNormalMembers();
     if (dsMembers.isEmpty()) {
-      result = ResultBuilder.createInfoResult(CliStrings.GATEWAY_MSG_MEMBERS_NOT_FOUND);
-    } else {
-      boolean gatewaySenderExists = false;
-      for (DistributedMember member : dsMembers) {
-        GatewaySenderMXBean bean;
-        if (cache.getDistributedSystem().getDistributedMember().getId().equals(member.getId())) {
-          bean = service.getLocalGatewaySenderMXBean(senderId);
-        } else {
-          ObjectName objectName = service.getGatewaySenderMBeanName(member, senderId);
-          bean = service.getMBeanProxy(objectName, GatewaySenderMXBean.class);
-        }
-        if (bean != null) {
-          gatewaySenderExists = true;
-          bean.rebalance();
-          GatewayCommandsUtils.accumulateStartResult(resultData, member.getId(),
-              CliStrings.GATEWAY_OK, CliStrings.format(
-                  CliStrings.GATEWAY_SENDER_0_IS_REBALANCED_ON_MEMBER_1, senderId, member.getId()));
-        } else {
-          GatewayCommandsUtils.accumulateStartResult(resultData, member.getId(),
-              CliStrings.GATEWAY_ERROR,
-              CliStrings.format(CliStrings.GATEWAY_SENDER_0_IS_NOT_AVAILABLE_ON_MEMBER_1, senderId,
-                  member.getId()));
-        }
+      return ResultModel.createInfo(CliStrings.GATEWAY_MSG_MEMBERS_NOT_FOUND);
+    }
+
+    ResultModel resultModel = new ResultModel();
+    TabularResultModel resultData = resultModel
+        .getTableSection(CliStrings.LOAD_BALANCE_GATEWAYSENDER);
+
+    boolean gatewaySenderExists = false;
+    for (DistributedMember member : dsMembers) {
```

Comment:
A better way to build TabularResultModel is to use its setColumnHeader and addRow methods. It gives the code a lot more readability and avoid using the utility class.